dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,142 @@
+
+### naming note
+
+- `isValid*` should mean that a full validity check is made
+- without `Valid` (like in the case of `isPentagon`), we do not guarantee
+  that a full validity check is made; instead, a user should assume only a
+  minimal bit mask check is done
+    - we could imagine a `isValidPengaton` function, if full verification
+      would be convenient
+- similarly, a function like `cellsAreNeighbors` will assume each cell
+  has passed the `isValidCell` check; the function will do only minimal
+  work to determine if they are neighbors
+
+### todo
+- Do we have general guidance on when to use a `get` prefix with a function and when not?
+
+
+## H3 Grid Functions
+
+| Current name |     Proposed name      |
+|--------------|------------------------|
+| `h3Distance` | `gridDistance`         |
+| `h3Line`     | `gridPathCell`         |
+| *DNE*        | `gridPathEdge`         |
+| *DNE*        | `gridPathDirectedEdge` |
+
+
+## H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
+
+
+todo: do we want/need to use `get` to prefix all these functions?
+
+## Area/Length Functions
+
+|  Current name  |    Proposed name     |                   Notes                    |
+|----------------|----------------------|--------------------------------------------|
+| `hexAreaKm2`   | `hexAreaAvgKm2`      | todo: `hexAreaMaxKm2` and `hexAreaMinKm2`? |
+| `hexAreaM2`    | `hexAreaAvgM2`       | todo: add min/max version?                 |
+| `edgeLengthKm` | `hexEdgeLengthAvgKm` | todo: add min/max version?                 |
+| `edgeLengthM`  | `hexEdgeLengthAvgM`  | todo: add min/max version?                 |
+| *DNE*          | `pentagonAreaAvgKm2` | plus others                                |
+
+todo: Could we add functions with signatures like `float cellAreaKm2(H3Index h)` to compute the actual area of a specific H3 cell?
+
+## Cell Neighborhood Functions
+
+| Current name |    Proposed names    |     Notes      |
+|--------------|----------------------|----------------|
+| `kRing`      | `disk` or `gridDisk` | filled-in disk |
+| `hexRing`    | `ring` or `gridRing` | hollow ring    |
```

Comment:
Note that `hexRing` is one of the hex-only functions, and I don't believe has a fallback in the C library.

Reply:
noted!